include/pybind11/stl_bind.h

```diff
@@ -569,6 +569,14 @@ template <typename Map, typename Class_> auto map_if_insertion_operator(Class_ &
     );
 }
 
+template <typename Map, typename Class_> auto map_if_equal_operator ( Class_ &cl)
+-> decltype(std::declval<typename Map::key_type>() == std::declval<typename Map::key_type>() and
+  std::declval<typename Map::mapped_type>() == std::declval<typename Map::mapped_type>(), void()) {
+
+    cl.def("__eq__", [](const Map& lhs, const Map& rhs) {
```

Comment:
Both `std::map` and `std::unordered_map` actually provide their own equality test (that is potentially more efficient). See https://en.cppreference.com/w/cpp/container/map/operator_cmp and https://en.cppreference.com/w/cpp/container/unordered_map/operator_cmp. It would be preferable to rely on these implementations rather than recreating an equality test here.